src/Content/Backend/Solution/Monaco.Template.Backend.Domain.Tests/FileTests.cs

```diff
@@ -0,0 +1,261 @@
+		sut.Size.Should().Be(size);
+		sut.ContentType.Should().Be(contentType);
+		sut.IsTemp.Should().Be(isTemp);
+		sut.UploadedOn.Should().Be(uploadedOn);
+	}
+
+	[Theory(DisplayName = "New File with empty name fails")]
+	[AutoDomainData]
+	public void NewFileWithEmptyNameFails(Guid id,
+										  string extension,
+										  long size,
+										  string contentType,
+										  bool isTemp,
+										  DateTime uploadedOn)
+	{
+		extension = extension[..File.ExtensionLength];
+
+		var sut = () => FileFactory.CreateMock((id,
+												string.Empty,
+												extension,
+												size,
+												contentType,
+												isTemp,
+												uploadedOn));
+
+		sut.Should()
+		   .Throw<Exception>()
+		   .WithInnerException<Exception>()
+		   .WithInnerException<ArgumentException>();
+	}
+
+	[Theory(DisplayName = "New File with name too large fails")]
+	[AutoDomainData]
+	public void NewFileWithNameTooLargeFails(Guid id,
+										  string extension,
+										  long size,
+										  string contentType,
+										  bool isTemp,
+										  DateTime uploadedOn)
+	{
+		extension = extension[..File.ExtensionLength];
+
+		var sut = () => FileFactory.CreateMock((id,
+												new string(It.IsAny<char>(),
+														   File.NameLength + 1),
+												extension,
+												size,
+												contentType,
+												isTemp,
+												uploadedOn));
+
+		sut.Should()
+		   .Throw<Exception>()
+		   .WithInnerException<Exception>()
+		   .WithInnerException<ArgumentException>();
+	}
+
+	[Theory(DisplayName = "New File with empty extension fails")]
+	[AutoDomainData]
+	public void NewFileWithEmptyExtensionFails(Guid id,
+											   string name,
+											   long size,
+											   string contentType,
+											   bool isTemp,
+											   DateTime uploadedOn)
+	{
+		var sut = () => FileFactory.CreateMock((id,
+												name,
+												string.Empty,
+												size,
+												contentType,
+												isTemp,
+												uploadedOn));
+
+		sut.Should()
+		   .Throw<Exception>()
+		   .WithInnerException<Exception>()
+		   .WithInnerException<ArgumentException>();
+	}
+
+	[Theory(DisplayName = "New File with extension too large fails")]
+	[AutoDomainData]
+	public void NewFileWithExtensionTooLargeFails(Guid id,
+											 string name,
+											 long size,
+											 string contentType,
+											 bool isTemp,
+											 DateTime uploadedOn)
+	{
+		var sut = () => FileFactory.CreateMock((id,
+												name,
+												new string(It.IsAny<char>(),
+														   File.ExtensionLength + 1),
+												size,
+												contentType,
+												isTemp,
+												uploadedOn));
+
+		sut.Should()
+		   .Throw<Exception>()
+		   .WithInnerException<Exception>()
+		   .WithInnerException<ArgumentException>();
+	}
+
+	[Theory(DisplayName = "New File with negative or zero size fails")]
+	[AutoDomainData]
+	public void NewFileWithNegativeOrZeroSizeFails(Guid id,
```

Comment:
Should this be split into separate tests? Would expect to have only 1 system under test per test?

Reply:
Thought the same, but since the validation method is only 1 (` .NegativeOrZero()`), then I thought a single test should be enough... What do you think?

Reply:
Never mind. Splitting it into 2 methods anyway.